src/targets/objc/native.js

```diff
@@ -1,65 +1,84 @@
 'use strict';
 
 var util = require('util');
+var objcHelpers = require('./helpers');
 
 module.exports = function (options) {
   var opts = util._extend({
-    timeout: '10'
+    timeout: '10',
+    indent: true
   }, options);
 
   var code = [];
 
-  // Dependencies
   code.push('#import <Foundation/Foundation.h>');
+
+  if (this.source.cookies.length) {
+    var cookies = this.source.cookies.map(function (cookie) {
+      return encodeURIComponent(cookie.name) + '=' + encodeURIComponent(cookie.value);
+    });
+    this.source.headersObj['Cookie'] = cookies.join('; ');
+  }
+
+  // Build headers (including cookies)
+  if (Object.keys(this.source.headersObj).length) {
+    code.push(null);
+    code.push(objcHelpers.nsDictionaryBuilder('headers', this.source.headersObj, opts.indent));
+  }
+
+  var parameters;
+
+  if (this.source.postData.mimeType && (this.source.postData.text || this.source.postData.params)) {
+    switch (this.source.postData.mimeType) {
+      case 'application/x-www-form-urlencoded':
+        parameters = this.source.postData.params;
+
+        code.push(null);
```

Comment:
@ahmadnassri As already explained before, i have no choice but to do that, otherwise, an example: `headers.json`

```js
{
  "method": "GET",
  "url": "http://mockbin.com/har",
  "headers": [
    {
      "name": "Accept",
      "value": "application/json"
    },
    {
      "name": "X-Foo",
      "value": "Bar"
    }
  ]
}
```

Will enter the switch, and create a line:

```
NSData *postData = [[NSData alloc] initWithData:[@"undefined" dataUsingEncoding:NSUTF8StringEncoding]];
```

And actually, since I just updated the `plain/text` case with a `v.replace(/"/g, '\\"')`, we will now have an undefined error.